### ISSUE TITLE
terminate called after throwing an instance of 'xsd::cxx::tree::parsing<char>'

### ISSUE THREAD

**Issue by adam21** [DLR-TS/xodr#1]
Hello,

I managed to build the OpenDrive Driver with the GDAL base library following steps described [here](https://github.com/DLR-TS/gdal).

T downloaded some test datasets from the official OpenDrive download section. But when trying to convert .xodr file to ESRI Shapefile, I got the following message : 

`terminate called after throwing an instance of 'xsd::cxx::tree::parsing<char>'
 what():  instance document parsing failed
Aborted (core dumped)
`

I saw that you already mentioned the issue in [this](https://github.com/DLR-TS/gdal) page, is there any progress on this topic ? 

Thank you !

**Comment by michikommader** [comment]
This error is related to the XSD data binding library and occurs at least when the file path to your .xodr-file is invalid. Double-check the file path, it should be relative to the execution directory from which you are calling ogr2ogr.

**Comment by adam21** [comment]
I tried this and it always gave the same message. I'm not sure if I did it the right way.

Can you provide an example of how you set the file path of an .xodr ??

**Comment by adam21** [comment]
Now when running the following command:

`../bin/ogr2ogr -f "ESRI Shapefile" ../dataset/file02.shp ../dataset/KA-Suedtangente-atlatec-Roadshape.xodr`

I got the following message :

`Segmentation fault (core dumped)`

I think that the last message is related to GDAL. But I'm still not sure that I well-defined the location path of the .xodr file

**Comment by adam21** [comment]
After several tests, I think the problem is related to the paramPoly3 geometries. The conversion works fine with line, arc, spiral ... geometrie types. 

the GDAL error message `Segmentation fault (core dumped)` appears only when the input xodr file has paramPoly3 geometries.

**Comment by cristhianmurcia182** [comment]
Hi, the error appears when the paramPoly3 object does not contain the pRange attribute. The pRange attribute is optional, and the current version of the code located in this repository assumes that all the paramPoly3 instances have this attribute. The code has been corrected by checking whether the pRange value is available if that is not the case, we define this parameter by hand in the range [0;1], as stated in the OpenDrive 1.4. documentation. After making this changes, I executed the code, with the /KA-Suedtangente-atlatec-Roadshape.xodr, obtaining appropriate results. I will update the code soon so that this error does not occur again.
Kind Regards, sorry for the inconveniences.
